lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -706,8 +706,18 @@ struct ModuleNameManager {
   /// added.
   StringRef getName(ValueOrOp valueOrOp) {
     auto entry = nameTable.find(valueOrOp);
-    assert(entry != nameTable.end() &&
-           "value expected a name but doesn't have one");
+    if (entry == nameTable.end()) {
+      llvm::errs() << "Name: ";
+      if (auto v = valueOrOp.dyn_cast<Value>())
+        v.print(llvm::errs());
+      else
+        valueOrOp.get<Operation *>()->print(llvm::errs());
+      llvm::errs()
+          << " Not found in name table! Most likely indicates that the given "
+             "op did not have an emitter in ExportVerilog, and should have "
+             "been lowered away before reaching this point.";
+      assert(false);
```

Comment:
This approach makes sense to me. I agree with Mike that in the longer time it would be better to make `getName` function fallible. I'm also wondering we can emit an error for unsupported operations (i.e. non-HW/SV/Comb dialects operations) before reaching to the emission. I'm not sure what's the best place to emit error, probably we can detect unsupported dialect operations in PrepareForEmission.

Reply:
If this is the short term solution, please at least add a message to the assert like `assert(false && "name not found (see above error)")` or something. But +1 for catching unsupported ops earlier, so we don't have to change getName.